tests/integration/kubernetes/k8s-guest-pull-image.bats

```diff
@@ -9,30 +10,149 @@ load "${BATS_TEST_DIRNAME}/../../common.bash"
 load "${BATS_TEST_DIRNAME}/tests_common.sh"
 
 setup() {
-    [[ "${PULL_TYPE}" =~  "guest-pull" ]] ||  skip "Test only working for pulling image inside the guest"
+    [[ "${PULL_TYPE}" =~ "guest-pull" ]] ||  skip "Test only working for pulling image inside the guest"
     setup_common
+    get_pod_config_dir
+    unencrypted_image_1="quay.io/prometheus/busybox:latest"
+    unencrypted_image_2="alpine:latest"
 }
 
-@test "Test can pull an unencrypted image inside the guest" {
-    pod_config="$(new_pod_config quay.io/prometheus/busybox:latest "kata-${KATA_HYPERVISOR}")"
+@test "Test we can pull an unencrypted image (busybox) outside the guest with runc and then inside the guest successfully" {
+    # Create one runc pod with busybox image
+    runc_pod_config="$(new_pod_config "$unencrypted_image_1" "kata-${KATA_HYPERVISOR}")"
+    sed -i '/runtimeClassName:/d' $runc_pod_config
+    echo "    command: [\"sleep\", \"30\"]" >> $runc_pod_config
 
-    kubectl create -f "${pod_config}"
+    # For debug sake
+    echo "Pod $runc_pod_config file:"
+    cat $runc_pod_config
 
-    # Get pod specification
-    kubectl wait --for=condition=Ready --timeout=$timeout pod "test-e2e"
+    set_node "$runc_pod_config" "$node"
```

Comment:
If we need to do set_node for all our pod_configs, should we consider integrating it into `new_pod_config` and pass it as a third parameter to check the test files cleaner?

Reply:
Part of me is wondering about adding metadata_annotations to the `new_pod_config` too, but I can see scenarios, like in this file that they will need to be optional, so that would make it need to be the last parameter. Maybe still worth considering though.

Reply:
I tried to move `$node` to `new_pod_config` function but failed,  because the `runtimeClassName` is one optional parameter too one function can't have 2+ optional parameters.
I didn't make the `$node` as a required parameter, I assume that we do NOT need set in for all other test cases?
So just leave the `set_node` call asis

Reply:
@wainersm - any thoughts on this. I think you added the node call as we are running these tests on AKS now, so getting the logs requires us to know which node it's running on, so do you think it should be optional? At the moment the only usage of new_pod_config is the k8s-measured-rootfs.bats test which also calls set_node

Reply:
I moved the `set_node` to `new_pod_config` function, the codes looks better than before.